konva.js

```diff
@@ -5118,6 +5118,113 @@ var Konva = {};
 })();
 
 (function () {
+    /**
+     * RGBA Filter
+     * @function
+     * @name RGBA
+     * @memberof Konva.Filters
+     * @param {Object} imageData
+     * @author codefo
+     * @example
+     * node.cache();
+     * node.filters([Konva.Filters.RGBA]);
+     * node.blue(120);
+     * node.green(200);
+     * node.alpha(0.3);
+     */
+    Konva.Filters.RGBA = function (imageData) {
+        var data = imageData.data,
+            nPixels = data.length,
+            red = this.red(),
+            green = this.green(),
+            blue = this.blue(),
+            alpha = this.alpha(),
+            i, ia;
+
+        for (i = 0; i < nPixels; i += 4) {
+            ia = 1 - alpha;
+
+            data[i] = red * alpha + data[i] * ia; // r
+            data[i + 1] = green * alpha + data[i + 1] * ia; // g
+            data[i + 2] = blue * alpha + data[i + 2] * ia; // b
```

Comment:
I (may be wrong) but I believe `data[i + 3]` contains the value for alpha component, so instead of multiplying RGB with alpha component, you could simply assign `data[i + 3] = ia;`